--- a/nv_ingest_client/util/vdb/milvus.py
+++ b/nv_ingest_client/util/vdb/milvus.py
@@ -691,8 +691,9 @@
 def add_metadata(element, meta_dataframe, meta_source_field, meta_data_fields):
     element_name = element["metadata"]["source_metadata"]["source_name"]
     df = meta_dataframe[meta_dataframe[meta_source_field] == element_name]
-    if df is None:
+    if df.empty:
         logger.info(f"NO METADATA ENTRY found for {element_name}")
+        return
     if df.shape[0] > 1:
         logger.info(f"FOUND MORE THAN ONE metadata entry for {element_name}, will use first entry")
     meta_fields = df[meta_data_fields]
